src/common/tools/gen_listings_and_webworkers.ts

```diff
@@ -78,12 +78,62 @@ export const listing = ${JSON.stringify(listing, undefined, 2)};
         `\
 // AUTO-GENERATED - DO NOT EDIT. See ${myself}.
 
-// oldG is a TestGroup<Fixture> object (defined in common/internal/test_group.ts).
-import { g as oldG } from '${relPathToSuiteRoot}/${entry.file.join('/')}.spec.js';
+// g is a TestGroup<Fixture> object (defined in common/internal/test_group.ts).
+import { g } from '${relPathToSuiteRoot}/${entry.file.join('/')}.spec.js';
+
+import { globalTestConfig } from '/out/common/framework/test_config.js';
+import { Logger } from '/out/common/internal/logging/logger.js';
+import { setDefaultRequestAdapterOptions } from '/out/common/util/navigator_gpu.js';
+import { parseQuery } from '/out/common/internal/query/parseQuery.js';
+import { comparePublicParamsPaths, Ordering } from '/out/common/internal/query/compare.js';
+import { assert } from '/out/common/util/util.js';
+
+async function reportTestResults(ev) {
+  const query = ev.data.query;
+  const expectations = ev.data.expectations;
+  const ctsOptions = ev.data.ctsOptions;
+
+  const { debug, unrollConstEvalLoops, powerPreference, compatibility } = ctsOptions;
+  globalTestConfig.unrollConstEvalLoops = unrollConstEvalLoops;
+  globalTestConfig.compatibility = compatibility;
+
+  Logger.globalDebugMode = debug;
+  const log = new Logger();
+
+  if (powerPreference || compatibility) {
+    setDefaultRequestAdapterOptions({
+      ...(powerPreference && { powerPreference }),
+      // MAINTENANCE_TODO: Change this to whatever the option ends up being
+      ...(compatibility && { compatibilityMode: true }),
+    });
+  }
+
+  const testQuerySingleCase = parseQuery(query);
+  let testcase = null;
+  for (const t of g.iterate()) {
```

Comment:
Between these two lines, `continue` the loop if the test name doesn't match, so we don't have to iterate all the cases of every test in the file

Reply:
Done. Note that I had to export `comparePath()`